pandas/core/generic.py

```diff
@@ -2276,7 +2276,7 @@ def to_json(
         date_format: str | None = None,
         double_precision: int = 10,
         force_ascii: bool_t = True,
-        date_unit: str = "ms",
+        date_unit: Literal["s", "ms", "us", "ns"] = "ms",
```

Comment:
this one seems broadly useful - perhaps let's define it as `TimeUnit` in `_typing.py` and import it here?